libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/SpanCreationContext.java

```diff
@@ -46,4 +105,36 @@ public String getSpanName() {
     public Attributes getAttributes() {
         return attributes;
     }
+
+    /**
+     * Returns the span kind.
+     * @return spankind.
+     */
+    public SpanKind getSpanKind() {
+        return spanKind;
+    }
+
+    /**
+     * Returns the parent span
+     * @return parent.
+     */
+    public SpanContext getParent() {
+        return parent;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Same question, under what circumstances the `equals` and `hashCode` is needed?  Those could not be implemented properly using mutable fields anyway